amethyst_renderer/src/pass/flat_colored/separate.rs

```diff
@@ -0,0 +1,240 @@
+//! Simple flat forward drawing pass.
+
+use gfx::pso::buffer::ElemStride;
+use gfx_core::state::{Blend, ColorMask};
+use glsl_layout::Uniform;
+
+use amethyst_assets::AssetStorage;
+use amethyst_core::{
+    ecs::prelude::{Join, Read, ReadExpect, ReadStorage},
+    transform::GlobalTransform,
+};
+use amethyst_error::Error;
+use derivative::Derivative;
+
+use crate::{
+    cam::{ActiveCamera, Camera},
+    hidden::{Hidden, HiddenPropagate},
+    mesh::{Mesh, MeshHandle},
+    mtl::{Material, MaterialDefaults},
+    pass::{
+        skinning::{create_skinning_effect, setup_skinning_buffers},
+        util::{draw_mesh, get_camera, VertexArgs},
+    },
+    pipe::{
+        pass::{Pass, PassData},
+        DepthMode, Effect, NewEffect,
+    },
+    skinning::JointTransforms,
+    tex::Texture,
+    types::{Encoder, Factory},
+    vertex::{Attributes, Color, Position, Separate, VertexFormat},
+    visibility::Visibility,
+    Rgba,
+};
+
+use super::*;
+
+static ATTRIBUTES: [Attributes<'static>; 2] = [
+    Separate::<Position>::ATTRIBUTES,
+    Separate::<Color>::ATTRIBUTES,
+];
+
+/// Draw mesh without lighting
+///
+/// See the [crate level documentation](index.html) for information about interleaved and separate
+/// passes.
+///
+/// # Type Parameters
+///
+/// * `V`: `VertexFormat`
+#[derive(Derivative, Clone, Debug, PartialEq)]
+#[derivative(Default(bound = "Self: Pass"))]
+pub struct DrawFlatColoredSeparate {
+    skinning: bool,
+    transparency: Option<(ColorMask, Blend, Option<DepthMode>)>,
+}
+
+impl DrawFlatColoredSeparate
+where
+    Self: Pass,
+{
+    /// Create instance of `DrawFlatColored` pass
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    /// Enable vertex skinning
+    pub fn with_vertex_skinning(mut self) -> Self {
+        self.skinning = true;
+        self
+    }
+
+    /// Enable transparency
+    pub fn with_transparency(
+        mut self,
+        mask: ColorMask,
+        blend: Blend,
+        depth: Option<DepthMode>,
+    ) -> Self {
+        self.transparency = Some((mask, blend, depth));
+        self
+    }
+}
+
+impl<'a> PassData<'a> for DrawFlatColoredSeparate {
+    type Data = (
+        Read<'a, ActiveCamera>,
+        ReadStorage<'a, Camera>,
+        Read<'a, AssetStorage<Mesh>>,
+        Read<'a, AssetStorage<Texture>>,
+        ReadExpect<'a, MaterialDefaults>,
+        Option<Read<'a, Visibility>>,
+        ReadStorage<'a, Hidden>,
+        ReadStorage<'a, HiddenPropagate>,
+        ReadStorage<'a, MeshHandle>,
+        ReadStorage<'a, Material>,
+        ReadStorage<'a, GlobalTransform>,
+        ReadStorage<'a, JointTransforms>,
+        ReadStorage<'a, Rgba>,
+    );
+}
+
+impl Pass for DrawFlatColoredSeparate {
+    fn compile(&mut self, effect: NewEffect<'_>) -> Result<Effect, Error> {
+        use std::mem;
+        let mut builder = if self.skinning {
+            create_skinning_effect(effect, FRAG_SRC)
```

Comment:
Basic vertex shader with skinning uses other vertex attributes - texture coordinates instead of color.